### PR TITLE
Enhancement analysis testing

### PR DESCRIPTION
**Pull request by mstapylton**
# Pull Request Template for MMEDS
Currently, I'm only testing one analysis, study setup. I know we want to test other demultiplexing options but it'd also be helpful to connect the setup of generating study files, analysis files (like creating the Qiime2_0 directory) with then running the study through github actions. Testing each study will add ~10mins runtime to test so we also might want to split up the github actions job into several that run concurrently as Adam suggested. I think it's best to handle all those things in a separate PR and have this PR focused on the initial changes needed for analysis testing. 

## What has changed
To automate running analyses, we needed to be able to run them through github actions which involved: 
1. automating installing all required packages, see tests.yml and also changed conda env files, changed module files. 
2. Created an example analysis by uploading test files to MMEDs, then modifying the study, analysis files to be run through github actions
3. Analyses typically run asynchronously via the watcher that hands off the job to lsf submission. To run the analysis via github actions, there needed to be some modifications to allow the analysis to run synchronously. Mostly this is handled through the new function in utils.py, run_analysis() but also through some changes to the job files. However, the changes to the jobfiles are just to consolidate the qiime2, jupyter, latex commands so that they can be run from mmeds subprocess calls, which is maybe a good thing.

Next steps: Add more analyses, automate creation of a new uploaded study from a "user" upload 

## Checklist of pre-requisites
-   [x] Does the code run?  
-   [x] Does the code follow the repository style?  
-   [x] Is the code tested?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,3 +61,5 @@ To install Qiime1.9.1 with a recent version of conda run:
 `conda config --set allow_conda_downgrades true`  
 `conda install conda=4.6.14`  
 Prior to running the conda command for installing Qiime
+
+#test
```
[the remark this answers is not in the document]
Comment (mstapylton) [reply README.md]: yes but I created a new readme in the .github/workflows directory

File: mmeds/tests/unit/test_analysis.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        cls.test_study = fig.TEST_STUDY
+
+    def test_qiime2(self):
+        """ Test running a qiime2 analysis """
+        # run_analysis() executes analyses synchronously rather than submitting as a job
+        run_analysis(self.test_study, 'qiime2', testing=True)
+        summarize_qiime(f'{self.test_study}/Qiime2_0', 'qiime2', testing=True)
+
+        # check for summary pdf, also gets uploaded as an artifact in github actions for inspection
+        pdf_output = Path(f'{self.test_study}/Qiime2_0/summary/mkstapylton@gmail.com-mattS-qiime2.ipynb')
```
[the remark this answers is not in the document]
Comment (mstapylton) [reply mmeds/tests/unit/test_analysis.py:25]: I uploaded a study via MMEDs using its test files, and then use that study's files for the analysis testing. So it's set in: mmeds-meta/test_files/test_study

**Review by adamcantor22 — APPROVED** [review]
now this PRs fate is in the hands of the github actions gods

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -14,28 +14,58 @@ jobs:
...
+
       - name: Start Services
         run: sudo systemctl start mysql.service; sudo systemctl start mongod.service;
+
       - name: Mysql Setup
         run: mysql -u root -proot < setup.sql
+
       - name: Server data setup
-        run: mkdir ~/mmeds_server_data; mkdir -p ~/conda_env_files/pheniqs; mkdir -p ~/.modules/modulefiles; cp -r ./test_files ~/mmeds_server_data/.; cp ./mmeds/resources/mmeds_stats.yaml ~/mmeds_server_data/.; cp -r ./modules/* ~/.modules/modulefiles/.; cp -r ./conda_env_files/* ~/conda_env_files/.;
+        run: mkdir ~/mmeds_server_data; mkdir ~/conda_envs; mkdir -p ~/conda_env_files/pheniqs; mkdir -p ~/.modules/modulefiles; cp -r ./test_files ~/mmeds_server_data/.; cp ./mmeds/resources/mmeds_stats.yaml ~/mmeds_server_data/.; cp -r ./modules/* ~/.modules/modulefiles/.; cp -r ./conda_env_files/* ~/conda_env_files/.;
```
[the remark this answers is not in the document]
Comment (cleme) [reply .github/workflows/tests.yml]: Agree, as mentioned in some other comment we should avoid `~` whenever possible and replace by either absolute path *or* use a variable that points to the root of MMEDS account/path.

[the remark this answers is not in the document]
Comment (mstapylton) [reply .github/workflows/tests.yml]: changed ~ to /home/runner for github actions

**Pull request merged**